app/src/components/map/Share.jsx

```diff
@@ -5,31 +5,111 @@ class Share extends Component {
 
   constructor(props) {
     super(props);
+
     this.state = {
-      loading: true
+      copied: false,   // If the input's content has been copied
+      copyError: false // If the copy action failed
     };
   }
 
+  /**
+   * Copy the input's content into the clipboard and set the state's "copied"
+   * attribute to true. If the action failed, set the "copyError" attribute to
+   * true. After few milliseconds, the "copied" attribute is resetted to false
+   * to display a quick visual feedback.
+   *
+   * @param {object} e - click event on the button
+   */
   onCopy(e) {
     e.preventDefault();
-    /* TODO */
+
+    this.refs.input.select();
+
+    let error = false;
+    try {
+      if (!document.execCommand('copy')) error = true;
+    } catch (err) {
+      error = true;
+    }
+
+    this.setState({
+      copied: !error,
+      error
+    });
+
+    /* We want the copied state to only last few millisecond to display a visual
+     * feedback */
+    if (!error) setTimeout(() => this.setState({ copied: false }), 1000);
+  }
+
+  /**
+   * Return the URL of the page with the workspace id included
+   *
+   * @returns {string} URL
+   */
+  getURLWithWorkspace() {
+    return `${location.origin}${location.pathname}?workspace=${this.props.workspaceId}`;
   }
 
   render() {
+    if (this.props.error) {
+      return (
+        <div className={styles['c-share']}>
+          <h2 className={styles.title}>Share this campaign</h2>
+          <p>
+            Sorry, an error prevented the workspace to be saved. Try again.
```

Comment:
Maybe something a bit friendlier, like "please login first to save this workspace" ?